File: uri/1829.py
from math import log, factorial, floor

def pow(a,b):
	k = a
	for i in xrange(b-1):
		k *= a
	return k

n = int(raw_input())
pedro_win = [False]*n

for i in xrange(n):
	try:
		a, b = map(int,raw_input().split('^'))
		c = int(raw_input()[:-1])
	except(EOFError):
		break
	power = floor(log(int(pow(a,b)))) 
	facto = floor(log(factorial(c)))
	print facto, power
	if power < facto:
		pedro_win[i] = True
		
lucas = pedro_win.count(False)
pedro = pedro_win.count(True)

if pedro > lucas:
	print "Campeao: Pedro!"
if lucas > pedro:
	print "Campeao: Lucas!"
else:
	print "A competicao terminou empatada!"
for i, x in enumerate(pedro_win):
	if x:
		print "Rodada #%d: Pedro foi o vencedor" % (i+1)
	else:
		print "Rodada #%d: Lucas foi o vencedor" % (i+1)

File: uri/1829.cpp
#include <iostream>
#include <stdio.h>
#include <math.h>
#include <string.h>

using namespace std;

/* Pedro gives c!, Lucas gives pow(a,b)
   Returns true if pedro wins */
bool solve(int a, int b, int c) 
{
	float power = 0, facto = 0;
	for (int i = 0; i < b; i++)
		power += log10(a);
	for (int i = 2; i <= c; i++)
		facto += log10(i);
	return facto > power;
}

int main()
{
	int n;
	cin >> n;
	bool pedro_wins[n];
	int pedro, lucas;
	memset(pedro_wins, false, n+1);
	for (int i = 1; i <= n; i++)
	{
		int a, b, c;
		scanf("%d^%d", &a, &b);
		scanf("%d!", &c);
		pedro_wins[i] = solve(a,b,c);
		if(pedro_wins[i]) pedro++;
		else lucas++;
	}

	if (pedro > lucas) cout << "Campeao: Pedro!" << endl;
	else if (lucas > pedro) cout << "Campeao: Lucas!" << endl;
	else cout  << "A competicao terminou empatada!" << endl;

	for (int i = 1; i <= n; ++i)
		printf("Rodada #%d: %s foi o vencedor\n", i, (pedro_wins[i]) ? "Pedro" : "Lucas");
	

	return 0;
}